TGEngine/gamecontent/Actor.hpp

```diff
@@ -8,31 +8,67 @@
 namespace tge::gmc {
 
 	/*
-     * This is struct holds all the the properties that
+     * This is struct holds all the properties that
 	 * do not directly influence the mesh it self
      *
      * <ul>
      * <li><strong class='atr'>localTransform</strong> is the local tranform of the actor that will be computed at runtime</li>
      * <li><strong class='atr'>material</strong> is the index of the material used to draw this mesh</li>
-     * <li><strong class='atr'>layer</strong> is the layer the actor is drawn on (0 for world or 1 for UI)</li></ul>
+     * <li><strong class='atr'>layer</strong> is the layer the actor is drawn on (e.g. 0 for normal world)</li></ul>
      *
      * <h4>Valid usage</h4>
      * <ul>
-     * <li><strong class='atr'>localTransform</strong> must be a non-zero matrix</li>
+     * <li><strong class='atr'>localTransform</strong> must be a non-zero matrix otherwise this actor will be ignored</li>
      * <li><strong class='atr'>material</strong> must be a valid material id in the material list</li>
-     * <li><strong class='atr'>material</strong> must be accessable in the given layer</li>
-     * <li><strong class='atr'>layer</strong> must be a valid layer id</li></ul>
+     * <li><strong class='atr'>layer</strong> must be a valid layer id otherwise this actor will be ingnored</li></ul>
      */
 	struct ActorProperties {
 		glm::mat4 localTransform;
-		uint8_t   material;
-		uint8_t   layer;
+		uint32_t  material;
```

Comment:
i don't quite understand why you changed this to be a regular int when the limit of 256 materials hasn't changed

Reply:
To make it short: Memory alignment.